internal/gitinterface/log_test.go

```diff
@@ -102,4 +108,171 @@ func TestGetCommitsBetweenRange(t *testing.T) {
+		}
+
+		treeHash, err := WriteTree(repo, objects)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		treeHashes = append(treeHashes, treeHash)
+	}
+
+	// creating the first commit
+
+	commit := CreateCommitObject(testGitConfig, treeHashes[0], []plumbing.Hash{ref.Hash()}, fmt.Sprintf("Test commit %v", 1), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// creating two children from the first commit
+
+	children := make([]plumbing.Hash, 0, 2)
+	for i := 1; i <= 2; i++ {
+		commit := CreateCommitObject(testGitConfig, treeHashes[i], []plumbing.Hash{ref.Hash()}, fmt.Sprintf("Test commit %v", i+1), testClock)
+		childRefName := plumbing.ReferenceName(fmt.Sprintf("refs/heads/child%d", i))
+		childRef, err := applyCommitAndRefreshRef(repo, commit, plumbing.NewHashReference(childRefName, plumbing.ZeroHash))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		commitIDs = append(commitIDs, childRef.Hash())
+		children = append(children, childRef.Hash())
+	}
+	// creating a merge commit from the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[4], children, fmt.Sprintf("Test commit %v", 5), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// creating children for each of the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[3], []plumbing.Hash{children[0]}, fmt.Sprintf("Test commit %v", 4), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[5], []plumbing.Hash{children[1]}, fmt.Sprintf("Test commit %v", 6), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// Git tree with merge commit structure without its commit trees and its values :
+	//
+	//  commit 4       commit 5         commit 6
+	//    │              │  │              │
+	//    └─► commit 2 ◄─┘  └─► commit 3 ◄─┘
+	// 		      │              │
+	// 		 	  └─► commit 1 ◄─┘
+
+	allCommits, err := getCommitsFromCommitIds(commitIDs, repo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Run("Get all commits from merge commit 5", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[3], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		expectedCommits := []*object.Commit{allCommits[3], allCommits[2], allCommits[1], allCommits[0]}
+		sort.Slice(expectedCommits, func(i, j int) bool {
+			return expectedCommits[i].ID().String() < expectedCommits[j].ID().String()
+		})
+
+		assert.Equal(t, expectedCommits, commits)
+		assert.Nil(t, err)
+	})
+	t.Run("Get all commits from merge commit 6", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[3], plumbing.ZeroHash)
```

Comment:
This invocation looks identical to the one for merge commit 5

Reply:
Also, is 6 a merge commit? In the diagram its parent seems to be commit 3 only?

Reply:
Yes, this is the wrong test name, I was thinking about something on the line of, "Get commits from leaf commit 6 to root commit 1"

Reply:
Fixed, changed the test name, and fixed the test to actually reflect test name. Merge commit 5's test has also been fixed, before it was not actually testing the merge commit, but something else.